content/cpp/concepts/smart-pointers/smart-pointers.md

```diff
@@ -0,0 +1,82 @@
+---
+Title: 'Smart Pointers'
+Description: 'A smart pointer is a class that holds and manages a pointer through RAII.'
+Subjects:
+  - 'Computer Science'
+  - 'Game Development'
+Tags:
+  - 'Smart Pointers'
+  - 'Pointers'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+## Smart Pointers
+
+A smart pointer is a class that holds and manages a pointer through RAII. It allows developers to not worry about freeing a pointer, and it also allows pointers to be exception-safe.
+
+Key Concept:
+
+ - Ownership: When refering to memory, ownership answers the question: 'Who is in charge of freeing the given resource?'
+
+There are three standard smart pointers defined in the `<memory>` header file:
+
+ - `std::unique_ptr<T>` A unique pointer does not share ownership, and will free the resource at the end of the scope.
+ - `std::shared_ptr<T>` A shared pointer does share ownership, and will only free the resource when there are no other owners counted and it has reached the end of the scope.
+ - `std::weak_ptr<T>` A weak pointer is used with a shared_pointer, but it does not add to the reference counter like a shared pointer does.
+
+Both `unique_ptr` and `shared_ptr` have a corresponding function to create their respective pointers:
+
+ - `std::make_unique<T>(value)`
+ - `std::make_shared<T>(value)` 
+
+
+## Example
+
+Unique Pointers:
+
+```cpp
+int main()
+{
+    auto ptr = std::make_unique<int>(10);
+} // ptr reaches end of scope, no memory leaks
+```
+
+Shared Pointers:
+
+```cpp
+int main()
+{
```

Comment:
For Codecademy's C++ content, the opening curly bracket stays on the same line:

```
int main() {

}
```

Reply:
Will do, thanks for all the feedback!